### PR TITLE
osc: Fix and simplify limited_list

### PR DESCRIPTION
**Pull request by mkoskar**
I agree that my changes can be relicensed to LGPL 2.1 or later.

* Fixes: when on the end of playlist only half of entries are displayed.
* Simplifies the logic of limited_list so it's easy to follow.
* limited_list's pos parameter is now 1 based which seem more natural.
* Few changes to comply with code style thorough the file.
* Small format change:

  "Playlist: (%d/%d):"  -> "Playlist [%d/%d]:"
  "Chapters: (%d/%d):"  -> "Chapters [%d/%d]:"

### CODE REVIEW

File: player/lua/osc.lua [hunk 1]
```diff
@@ -651,47 +651,44 @@ end
...
+        item.current = (i == pos) and true or nil
+        table.insert(reslist, item)
     end
-    return count, temp
+    return count, reslist
 end
 
 function get_playlist()
-    local pos = mp.get_property_number('playlist-pos')
+    local pos = mp.get_property_number('playlist-pos') + 1
```
Comment (wiiaboo) [inline player/lua/osc.lua]: There's a `playlist-pos-1` property you can use instead. Not for chapter though.

Comment (mkoskar) [inline player/lua/osc.lua]: Yes, I know ... and had it there before. But for consistency with chapter I've changed it back :|.

Comment (wiiaboo) [inline player/lua/osc.lua]: Yeah, that makes sense too.

**Comment by wiiaboo** [comment]
Merged, thanks.

**Comment by mkoskar** [comment]
Thanks for merging ;)

**Pull request closed without merging**